tfsdk/resource_import.go

```diff
@@ -6,10 +6,26 @@ import (
 	"github.com/hashicorp/terraform-plugin-go/tftypes"
 )
 
+// Optional interface on top of Resource that enables provider control over
+// the ImportResourceState RPC. This RPC is called by Terraform when the
+// `terraform import` command is executed. Afterwards, the ReadResource RPC
+// is executed to allow providers to fully populate the resource state.
+type ResourceWithImportState interface {
```

Comment:
perhaps nit-picking, but shouldn't this interface contain `Resource`, which would allow for validation that an implementation is fully implementing both using a single line? e.g.

```
var _ ResourceWithImportState = MyResource{}
# which also implies
# var _ Resource = MyResource{}
```